Automattic/wp-calypso
# Framework: Convert NotificationsSettings store to redux & remove Immutable from module.

#### Changes proposed in this Pull Request

This PR started as an effort to remove ImmutableJS from Calypso (#28652), replacing it's usage in the NotificationsSettings store with plain JS. Doing that took us most of the way to replacing the entire flux store with redux state - one step closer to removing Immutable (there's now only one module left) and one step closer to removing flux altogether too.

#### Testing instructions

Visit each of the following links (you can use the tabs in Calypso) and try toggling some settings on each (they each use the different `toggleState` methods). 
Hit save and refresh the page to ensure that the settings have persisted.

http://calypso.localhost:3000/me/notifications
http://calypso.localhost:3000/me/notifications/comments
http://calypso.localhost:3000/me/notifications/updates

You should also notice either a success or error notice depending on the response of the save request (you can mock a failure by 'throttling' your network to mimick being offline).


### Notes:

#### `blogs`/`blog` in `toggleState`
`NotificationSettingsStore.getStateFor` gets called with the values `'other'`, `'wpcom'` and `'blogs'`  whereas `toggleState` has the methods `other`, `wpcom` and `blog`. This is accounted for by `toggleSetting`, which defaults to calling the `blog` method when a `source` is passed in that it doesn't have a method to match. I don't think that the `blog` method ever actually gets called directly, only via defaulting. I believe it likely that we can just rename the `blog` method to blogs and never have the need for defaulting - this would need investigation though, to be sure.
